### PR TITLE
Rustify examples and resolve Cargo issues

### PR DESCRIPTION
**Pull request by rekka**
Hi,

One more pull request:

- Move examples so that they match Cargo's standard layout. Now they can be run using `cargo run --example name`. They are also automatically build when running `cargo test`.

- Fix some issues when running `cargo test`: comments were misinterpreted as doc tests.

- Remove `Makefile`: I found out that `Makefile` simply does not seem work anymore (issues resolving external crates), so I removed it since all is now handled in a standard way by Cargo. I think this reduces confusion.

- Add `cargo test` to `.travis.yml` to build examples and in general run the test suite (once there is one).

- Update `README.md`: Cargo and crates.io are the way to use this library.

- (Bunch of trailing whitespace removed--my editor removes it by default.)

Please feel free to disregard this pull request if you do not approve of the changes. I simply like your project and I think it'd be great if it were up to date with the current version of Cargo.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,38 +7,31 @@ A __Rust__ binding for the [GSL library](http://www.gnu.org/software/gsl/) (the
 
 This binding requires the [GSL library](http://www.gnu.org/software/gsl/) library to be installed.
 
-To build it, please use :
+This crate works with Cargo and is on
+[crates.io](https://crates.io/crates/GSL).
```
Comment (GuillaumeGomez) [inline README.md]: I'm just wondering: why so much backlines ? Here, you could have let "[crates.io]..." on the same line, no ?

File: README.md [hunk 2]
```diff
@@ -7,38 +7,31 @@ A __Rust__ binding for the [GSL library](http://www.gnu.org/software/gsl/) (the
...
-> make
+```toml
+[dependencies]
+GSL = "*"
 ```
 
-This command builds __rgsl__, the examples and the documentation.
-
-You can build them separatly too.
+Add the following line to your source:
```
Comment (GuillaumeGomez) [inline README.md]: "source code" instead of "source" maybe ? (I'm not an english native so correct me if I'm wrong !)

**Comment by GuillaumeGomez** [comment]
Thanks for this new PR ! Can you just check my comments ? After that I'll merge your work.

**Comment by rekka** [comment]
Thanks for checking it! I modified `README.md` as you suggested (also not a native speaker here). As for the text wrapping, it is pretty standard to wrap markdown to 72 characters. Works better with git tools. I changed the links to reference links, it looks prettier now.

**Comment by rekka** [comment]
I realized that the module documentation should also be updated.

File: README.md [hunk 3]
```diff
@@ -1,46 +1,41 @@
...
-file:///{rgsl_location}/doc/rgsl/index.html
+Then open this file with an internet browser:
+`file:///{rgsl_location}/target/doc/rgsl/index.html`
 
 You can also access the latest build of the documentation via the internet [here](http://rust-ci.org/GuillaumeGomez/rust-GSL/doc/rgsl/).
 
 ## License
 __rust-GSL__ is a wrapper for __GSL__, therefore inherits the [GPL license](http://www.gnu.org/copyleft/gpl.html).
+
+[crates.io]: https://crates.io/crates/GSL
```
Comment (GuillaumeGomez) [inline README.md]: I'm not sure to understand what you wanted to do here. In addition, it seems it doesn't render in github. Do you wanted to do something else ?

File: src/rgsl.rs [hunk 4]
```diff
@@ -2,54 +2,48 @@
...
+//! `file:///{rgsl_location}/target/doc/rgsl/index.html`
+//!
+//! You can also access the latest build of the documentation via the internet
+//! [here](http://rust-ci.org/GuillaumeGomez/rust-GSL/doc/rgsl/).
+//!
+//! ## License
+//! __rust-GSL__ is a wrapper for __GSL__, therefore inherits the
+//! [GPL license](http://www.gnu.org/copyleft/gpl.html).
+//!
+//! [crates.io]: https://crates.io/crates/GSL
```
Comment (GuillaumeGomez) [inline src/rgsl.rs]: These two lines will have to be changed, just like [here](https://github.com/GuillaumeGomez/rust-GSL/pull/18/files#r37693241).

**Comment by GuillaumeGomez** [comment]
I knew it was common to wrap at 80 characters, but I didn't know for the 72 characters limit. I learn every day. :)

I finished my review (see the two comments). Can you check ? Thanks again for your work !

File: README.md (hunk 3 above)
Comment (rekka) [inline README.md]: These are supposed to be markdown [reference links](http://spec.commonmark.org/0.21/#shortcut-reference-link). They do not render, but in the text you can invoke them using `[title][ref]` or `[ref]`, which is shorter I guess. Though I managed to mess up in the first line, fixing it now.

**Comment by rekka** [comment]
Somehow I see people wrapping markdown at 72 characters, which I kinda like. It works well in terminal (when diffs are involved).

File: README.md [hunk 5]
```diff
@@ -1,46 +1,41 @@
 rust-GSL [![Build Status](https://api.travis-ci.org/GuillaumeGomez/rust-GSL.png?branch=master)](https://travis-ci.org/GuillaumeGomez/rust-GSL) [![Gitter](https://badges.gitter.im/Join Chat.svg)](https://gitter.im/GuillaumeGomez/rust-GSL?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge)
 ========
 
-A __Rust__ binding for the [GSL library](http://www.gnu.org/software/gsl/) (the GNU Scientific Library).
+A __Rust__ binding for the [GSL library][] (the GNU Scientific Library).
 
 ##Installation
 
-This binding requires the [GSL library](http://www.gnu.org/software/gsl/) library to be installed.
+This binding requires the [GSL library] library to be installed.
```
Comment (GuillaumeGomez) [inline README.md]: Shouldn't it "[GSL library][]" here too ?

File: README.md [hunk 6]
```diff
@@ -1,46 +1,41 @@
...
 
-To build it, please use :
+This crate works with Cargo and is on [crates.io].  Just add the
```
Comment (GuillaumeGomez) [inline README.md]: Shouldn't it be "[crates.io][]" ?

File: README.md (hunk 3 above)
Comment (GuillaumeGomez) [inline README.md]: Oh nice ! I totally ignored this. Thanks for teaching me new things !

File: src/rgsl.rs [hunk 7]
```diff
@@ -2,54 +2,48 @@
...
-## License
-__rust-GSL__ is a wrapper for __GSL__, therefore inherits the [GPL licence](http://www.gnu.org/copyleft/gpl.html).
-
-Here is the list of all modules :
-!*/
+//! A __Rust__ binding for the [GSL library][] (the GNU Scientific Library).
+//!
+//! ##Installation
+//!
+//! This binding requires the [GSL library] library to be installed.
```
Comment (GuillaumeGomez) [inline src/rgsl.rs]: Same remark here.

File: src/rgsl.rs [hunk 8]
```diff
@@ -2,54 +2,48 @@
...
+//!
+//! This crate works with Cargo and is on [crates.io].  Just add the
```
Comment (GuillaumeGomez) [inline src/rgsl.rs]: Same remark here.

**Comment by rekka** [comment]
The reference link format allows both `[crates.io][]` and `[crates.io]`. But this doesn't work in the title since `[GSL library]` is followed by `(the GNU scientific library)`, which gets parsed together as one link with title `GSL library` and address `the GNU scientific library`. So here it is necessary to keep the `[GSL library][]` form.

I can add `[]` everywhere if you prefer. But the current version is according to markdown spec.

**Comment by GuillaumeGomez** [comment]
@rekka: No, I trust you (and I didn't this in markdown). So now everything's good. Thanks a lot for your work, I merge !

**Comment by rekka** [comment]
Thanks!

**Pull request merged**